clarity/src/vm/contexts.rs

```diff
@@ -1783,11 +1783,14 @@ impl<'a> LocalContext<'a> {
         }
     }
 
-    pub fn lookup_callable_contract(
-        &self,
-        name: &str,
-    ) -> Option<&(QualifiedContractIdentifier, TraitIdentifier)> {
-        self.function_context().callable_contracts.get(name)
+    pub fn lookup_callable_contract(&self, name: &str) -> Option<&TraitData> {
+        match self.callable_contracts.get(name) {
+            Some(found) => Some(found),
+            None => match self.parent {
+                Some(parent) => parent.lookup_callable_contract(name),
+                None => None,
+            },
+        }
```

Comment:
I think that one of the advantages of re-orienting trait instances/callable contracts as actual `Value`s is that it means they could be treated like normal `variables` in the `LocalContext`, that means they don't need any special treatment from the `let` binding. It forces the question of where callable contract "come from" to be answered with the `Value::Trait` constructor (which I think means that it can only happen when `apply`ing function arguments).

Reply:
Yes, I made changes to add to the `callable_contracts` map at each place where a new binding occurs, but it could make sense now to remove that map and just use the `variables`.